text/0202-events-api.md

```diff
@@ -0,0 +1,184 @@
+
+Note: This could, for example, be implemented by storing any mutable configuration in the EventEmitterProvider and having EventEmitter implementation objects have a reference to the EventEmitterProvider from which they were obtained. If configuration must be stored per-EventEmitter (such as disabling a certain EventEmitter), the EventEmitter could, for example, do a look-up with its name+version+schema_url in a map in the EventEmitterProvider, or the EventEmitterProvider could maintain a registry of all returned EventEmitters and actively update their configuration if it changes.
+
+The effect of associating a Schema URL with a EventEmitter MUST be that the telemetry emitted using the EventEmitter will be associated with the Schema URL, provided that the emitted data format is capable of representing such association.
+
+### EventEmitter
+
+The EventEmitter is responsible for creating Events.
+
+Note that EventEmitters should usually not be responsible for configuration. This should be the responsibility of the EventEmitterProvider instead.
+
+#### EventEmitter operations
+
+The EventEmitter MUST provide functions to:
+
+- Function named “logEvent” to create an Event with the provided event name and attributes. The event name provided should be inserted as an attribute with key “event.name”. It will override any attribute with the same key in the attributes passed.
+- Function named “recordException” to record an Exception as an Event. This is to facilitate recording an exception outside of a trace context for languages that already do not support recording an exception in a log message. This should work similar to [Record Exception](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#record-exception) in Trace API, with the following semantics
+  - This is a specialized variant of logEvent function, so for anything not specified here, the same requirements as for logEvent apply.
+  - The signature of the method is to be determined by each language and can be overloaded as appropriate. The method MUST record an exception as an Event with the conventions outlined in the exception semantic conventions document. The minimum required argument SHOULD be no more than only an exception object.
+  - If RecordException is provided, the method MUST accept an optional parameter to provide any additional event attributes (this SHOULD be done in the same way as for the AddEvent method). If attributes with the same name would be generated by the method already, the additional attributes take precedence.
+  - Note: RecordException may be seen as a variant of logEvent with additional exception-specific parameters and all other parameters being optional (because they have defaults from the [exception semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md)).
+
+A LogRecord representing exception event will look like this:
```

Comment:
nit: `A LogRecord representing exception event may look like this:`
As we haven't defined the conventions for exceptions yet, it would be better to have the language define this as a possible view as opposed to _the_ (will) look like.

Reply:
@MSNev the conventions for exceptions are already defined, please see - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/exceptions.md

Reply:
I changed the text to say `may`.